webserver/views/api.py

```diff
@@ -248,20 +250,30 @@ def _messybrainz_lookup(listens):
     return augmented_listens
 
 
-def _validate_listen(listen):
+def _validate_listen(listen, listen_type):
```

Comment:
Good point about the constants, I didn't pick that up. Though, I'm suspicious that checking for string membership in a set would be (much) slower than ints.